src/main/java/com/gluonhq/substrate/target/LinuxTargetConfiguration.java

```diff
@@ -131,6 +132,92 @@ public boolean link() throws IOException, InterruptedException {
         return super.link();
     }
 
+    @Override
+    public boolean install() throws IOException, InterruptedException {
+        if (!crossCompile) {
+            return super.install();
+        }
+
+        String remoteHostName = projectConfiguration.getRemoteHostName();
+        String remoteDir = projectConfiguration.getRemoteDir();
+        if (remoteHostName == null || remoteHostName.isEmpty() || remoteDir == null || remoteDir.isEmpty()) {
```

Comment:
Divide the if into 2 different if..blocks to give more concrete error messages to the developer.